### PR TITLE
Implement this expression

### PR DESCRIPTION
**Pull request by sanxiyn**
This is my first PR here!

After playing with `boa` for a while, I noticed `this` is unimplemented. I tried to implement it with as minimal changes as possible, so it may not fit the overall architecture. If so, just tell me how should I update.

### CODE REVIEW

File: src/lib/environment/environment_record_trait.rs [hunk 1]
```diff
@@ -60,6 +60,11 @@ pub trait EnvironmentRecordTrait: Debug + Trace + Finalize {
     /// Return true if it does and false if it does not.
     fn has_this_binding(&self) -> bool;
 
+    fn get_this_binding(&self) -> Value {
+        debug_assert!(self.has_this_binding());
```
Comment (jasonwilliams) [inline src/lib/environment/environment_record_trait.rs]: not sure if the assert is needed?
It won't even compile if a `has_this_binding()` isn't implemented

**Comment by jasonwilliams** [comment]
hmm not sure about this, `this` is just a variable like any other, and [we create it](https://github.com/jasonwilliams/boa/blob/master/src/lib/exec.rs#L328) before entering a function. How will we deal with this.something() with your change?

File: src/lib/environment/environment_record_trait.rs (hunk 1 above)
Comment (sanxiyn) [reply src/lib/environment/environment_record_trait.rs]: `has_this_binding` can return false for some environment records, although the interpreter should never call `get_this_binding` for them. I am asserting that invariant here.

**Comment by sanxiyn** [comment]
Okay I will add a test with constructors.

**Comment by sanxiyn** [comment]
Now with better understanding, I reimplemented this in #185.

**Pull request closed without merging**